packages/react-form-state/src/components/tests/Nested.tsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import faker from 'faker';
+import {mount} from 'enzyme';
+
+import {Input} from './components';
+import FormState from '../..';
+
+const ARBITRARY_SEED = 1337;
+
+describe('<Nested />', () => {
+  beforeEach(() => {
+    faker.seed(ARBITRARY_SEED);
+  });
+
+  it('passes field state into child function', () => {
+    const renderPropSpy = jest.fn(() => null);
+
+    const product = {
+      title: faker.commerce.productName(),
+      adjective: faker.commerce.productAdjective(),
+    };
+
+    mount(
+      <FormState initialValues={{product}}>
+        {({fields}) => {
+          return (
+            <FormState.Nested field={fields.product}>
+              {renderPropSpy}
+            </FormState.Nested>
+          );
+        }}
+      </FormState>,
+    );
+
+    expect(renderPropSpy).toBeCalledWith({
+      title: {
+        name: 'nodejs.title',
+        dirty: false,
+        // eslint-disable-next-line no-undefined
+        error: undefined,
+        initialValue: product.title,
+        value: product.title,
+        onChange: expect.any(Function),
+        onBlur: expect.any(Function),
+      },
+      adjective: {
+        name: 'nodejs.adjective',
+        dirty: false,
+        // eslint-disable-next-line no-undefined
+        error: undefined,
+        initialValue: product.adjective,
+        value: product.adjective,
+        onChange: expect.any(Function),
+        onBlur: expect.any(Function),
+      },
+    });
+  });
+
+  it("updates the top level FormState's array when an inner field is updated", () => {
+    const product = {title: faker.commerce.productName()};
+    const newTitle = faker.commerce.productName();
+
+    const renderPropSpy = jest.fn(({fields: {product}}: any) => {
+      return (
+        <>
+          <FormState.Nested field={product}>
+            {({title}: any) => {
+              return <Input label="title" {...title} />;
+            }}
+          </FormState.Nested>
+        </>
+      );
+    });
+
+    const form = mount(
+      <FormState initialValues={{product}}>{renderPropSpy}</FormState>,
+    );
+
+    form
+      .find(Input)
+      .props()
+      .onChange(newTitle);
+    form.update();
+
+    const {fields} = lastCallArgs(renderPropSpy);
+
+    expect(fields.product.value.title).toBe(newTitle);
```

Comment:
`expect(renderPropSpy).toHaveBeenLastCalledWith(expect.objectContaining(...))`

Reply:
I don't like that for single field comparisons when we already need `lastCallArgs` in this file anyway.

Reply:
That said I should change some of the other places to do that.